source/_posts/2015-09-03-hiring-apprentices.markdown

```diff
@@ -0,0 +1,82 @@
+But this meant that if we wanted to make an apprenticeship work for us, we needed  the apprentices to have a skill and experience level similar to someone with one full year of professional experience.
+
+But I didn't think it should actually take a year to gain that experience and those skills.
+
+It is all too common for people to have a first programming job where they get assigned a laundry list of bugs and work alone [spending lots of time](http://www.happybearsoftware.com/how-to-survive-your-first-year-as-a-programmer.html) on just a few small conceptual misunderstandings.
+In my own experience my first two jobs wasted a lot of time because the experienced people on the team were too busy to provide guidance on which things were worth learning and which things I should skip until later.
+I rarely got feedback on my projects unless they broke something in production.
+
+I'm not qualified to write a decent curriculum, but I think that we can deliberately make a lot of time for smart people to talk through what they are learning.
+I think that with this kind of feedback the apprentices will significantly outpace the rate at which I learned things as a new developer.
+I also think that by hiring apprentices in a group and giving them tasks as a group, they will do a lot of learning as a team.
+This will help prepare them for the very chaotic and collaborative environment on the MX engineering team, and also speed up their learning because they spend time teaching and learning from people with a similar experience level.
+
+Based on the two ideas above we planned a mentorship model where each apprentice has a mentor assigned.
+This direct mentor has between 1 and 3 years of professional experience and sets aside one hour each day to helping their apprentice.
+The apprentice should ask for the type of help that they think will be most helpful to them.
+They might ask to pair program, answer a specific programming question, figure out a bug or go on a walk and discuss a career in software.
+The point is to give as much control as possible to the apprentice and let the mentor provide feedback.
+We also have a "senior mentor" that changes each week.
+The senior mentor gives 90 minutes of their week to each apprentice with the same basic premise as the direct mentor.
+
+
+# Knowing The Audience
+
+We had a lot of discussions internally about who we wanted to bring in as apprentices.
+Would we consider hiring people with no programming experience at all?
+Did long-term potential mean more than their current skill set?
+What evidence of long-term potential were we looking for?
+
+The biggest concern that I had was that we might hire people as apprentices, spend 3 months of their time, 3 months of our time and not be able to teach them enough to be successful engineers on our team.
+That was pretty much my definition of failure.
```

Comment:
"That is my definition of failure." This is a power statement. Make it powerful.